### PR TITLE
Added visual effects

### PR DESCRIPTION
**Pull request by OLODiman11**
Added:
- Movement/rotation transitions (tweening)
- Row deletion particle effect

Holding left/rights button has a visual bug. Will fix after major refactoring.

https://github.com/OLODiman11/tetris/assets/88656576/82eefd29-5c23-4d1d-9cc5-4c1a2fdccf13

### CODE REVIEW

File: scenes/particles_test.tscn
Comment (S1rMekr) [inline scenes/particles_test.tscn:1]: Do we need this scene? I just can't if it's used somewhere in the project

**Review by S1rMekr — CHANGES_REQUESTED** [review]
Hello, _OLOLODiman11!_ I really appreciate your work but I have **some** questions if you don't mind.

1) Do we need s **particle_test.tscn**? I don't see if it's whether used.
2) It appears that we have a folder "**shape_previews**" with scripts. Maybe we should move them to the _scripts_ folder (It's more about the refactoring of all our code). What do you think?

File: scenes/particles_test.tscn
Comment (OLODiman11) [reply scenes/particles_test.tscn:1]: Maybe i'm stupid but your main comment doesn't have a reply button. About moving scripts. Maybe we should do it on the refactoring stage?

Comment (S1rMekr) [reply scenes/particles_test.tscn:1]: I agree with you!

**Review by S1rMekr — APPROVED** [review]
yea

**Pull request merged**